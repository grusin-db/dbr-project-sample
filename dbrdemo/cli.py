import argparse
import logging
from databricks.sdk import WorkspaceClient
from . import spark

logger = logging.getLogger(__name__)


def cli_foobar():
    parser = argparse.ArgumentParser(add_help=True, description="Sample CLI")

    parser.add_argument('--foo', default="foo", type=str, required=False, help="FOO text")
    parser.add_argument('--bar', default="bar", type=str, required=False, help="BAR text")

    args = parser.parse_args()

    w = WorkspaceClient()
    logger.info(f"Current cluster id: {w.config.cluster_id!r}")
    cluster = w.clusters.get(w.config.cluster_id)
    logger.info(f"Current cluster name: {cluster.cluster_name!r}")

    df = spark.createDataFrame([[args.foo, args.bar]], "foo string, bar string")
    df.show(truncate=False)
