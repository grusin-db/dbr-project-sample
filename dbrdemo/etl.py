from . import spark


def get_big_taxi_trips():
    df = spark.table('samples.nyctaxi.trips') \
        .filter('fare_amount > 100')

    return df.collect()
