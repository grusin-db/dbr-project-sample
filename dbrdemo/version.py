__version__ = '0.0.2a0+2024.4.3'
