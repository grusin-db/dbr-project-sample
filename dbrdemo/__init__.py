from databricks.labs.blueprint.logger import install_logger

install_logger()

import logging

logging.getLogger().setLevel(level=logging.CRITICAL)
logger = logging.getLogger('dbrdemo')
logger.setLevel(logging.DEBUG)

from .version import __version__

logger.info(f"Using dbrdemo version: {__version__}")

import datetime

from pyspark.dbutils import DBUtils
from pyspark.sql import SparkSession


def _get_spark_seession() -> SparkSession:
    try:
        # if running in real databricks
        from pyspark.sql import SparkSession
        spark = SparkSession.getActiveSession()
    except: # NOQA
        spark = None

    if not spark:
        logger.debug("Trying to aquire vscode/pytest databricks connect session...")
        from databricks.connect.session import DatabricksSession
        spark_unsupported_tz_map = {"CEST": "+01:00"}

        spark = DatabricksSession.builder.getOrCreate()
        now = datetime.datetime.now()
        local_now = now.astimezone()
        local_tz = local_now.tzinfo
        local_tzname = local_tz.tzname(local_now)
        local_tzname = spark_unsupported_tz_map.get(local_tzname) or local_tzname
        spark.conf.set('spark.sql.session.timeZone', local_tzname)

        # TODO: use ascii art to make this personalized just for your code!!! :)
        #       Open source tools for generating ascii art: https://itsfoss.com/ascii-art-linux-terminal/
        logger.warning("_________ _______  _______ _________   _______  _______  ______   _______ ")
        logger.warning("\\__   __/(  ____ \\(  ____ \\\\__   __/  (       )(  ___  )(  __  \\ (  ____ \\")
        logger.warning("   ) (   | (    \\/| (    \\/   ) (     | () () || (   ) || (  \\  )| (    \\/")
        logger.warning("   | |   | (__    | (_____    | |     | || || || |   | || |   ) || (__    ")
        logger.warning("   | |   |  __)   (_____  )   | |     | |(_)| || |   | || |   | ||  __)   ")
        logger.warning("   | |   | (            ) |   | |     | |   | || |   | || |   ) || (      ")
        logger.warning("   | |   | (____/\\/\\____) |   | |     | )   ( || (___) || (__/  )| (____/\\")
        logger.warning("   )_(   (_______/\\_______)   )_(     |/     \\|(_______)(______/ (_______/")

    spark.conf.set('spark.sql.legacy.timeParserPolicy', 'CORRECTED')
    spark.conf.set('spark.sql.legacy.json.allowEmptyString.enabled', 'true')

    return spark


spark = _get_spark_seession()
dbutils = DBUtils(spark)
