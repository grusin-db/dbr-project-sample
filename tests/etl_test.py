from dbrdemo.etl import get_big_taxi_trips
from dbrdemo.test_utils import smart_verify

def test_taxis():
    df = get_big_taxi_trips()
    data = df.collect()

    assert len(data) == 7

def test_taxis_data():
    df = get_big_taxi_trips()

    smart_verify(df)



