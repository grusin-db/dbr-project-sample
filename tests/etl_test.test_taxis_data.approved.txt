|    | tpep_pickup_datetime   | tpep_dropoff_datetime   |   trip_distance |   fare_amount |   pickup_zip |   dropoff_zip |
|---:|:-----------------------|:------------------------|----------------:|--------------:|-------------:|--------------:|
|  0 | 2016-01-04 18:58:23    | 2016-01-04 18:58:45     |            0    |           105 |         7114 |          7114 |
|  1 | 2016-01-16 18:09:15    | 2016-01-16 18:09:23     |            0    |           105 |         7310 |          7310 |
|  2 | 2016-01-28 17:36:17    | 2016-01-28 18:27:48     |           21.3  |           115 |        10017 |         11042 |
|  3 | 2016-01-30 22:28:42    | 2016-01-30 22:30:14     |            0    |           188 |         7974 |          7974 |
|  4 | 2016-02-12 20:55:19    | 2016-02-12 21:52:38     |           20.85 |           275 |        10013 |          7008 |
|  5 | 2016-02-17 22:23:14    | 2016-02-17 23:06:15     |           25.46 |           130 |        10103 |          7901 |
|  6 | 2016-02-29 12:16:16    | 2016-02-29 12:16:53     |            0    |           260 |         8876 |          8876 |
